pkg/discovery/dtofactory/node_pools_group_dto_builder.go

```diff
@@ -0,0 +1,71 @@
+package dtofactory
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/turbonomic/kubeturbo/pkg/discovery/repository"
+	"github.com/turbonomic/kubeturbo/pkg/discovery/util"
+	"github.com/turbonomic/turbo-go-sdk/pkg/builder/group"
+	"github.com/turbonomic/turbo-go-sdk/pkg/proto"
+)
+
+type NodePoolsGroupDTOBuilder struct {
+	cluster  *repository.ClusterSummary
+	targetId string
+}
+
+func NewNodePoolsGroupDTOBuilder(cluster *repository.ClusterSummary,
+	targetId string) *NodePoolsGroupDTOBuilder {
+	return &NodePoolsGroupDTOBuilder{
+		cluster:  cluster,
+		targetId: targetId,
+	}
+}
+
+func (builder *NodePoolsGroupDTOBuilder) Build() []*proto.GroupDTO {
+	nodePools := builder.getNodePools()
+	if len(nodePools) == 0 {
+		glog.V(3).Infof("No node pools detected.")
+		return nil
+	}
+
+	var groupDTOs []*proto.GroupDTO
+	for poolName, members := range nodePools {
+		var dtos []*proto.GroupDTO
+		var err error
```

Comment:
Nit: We could simply remove this line as the err can be set below by:
```
dto, err := xxx
```

Reply:
Please update this.